### ISSUE TITLE
Bastok Mission 2-3 San d'Oria > Windurst Path - Dark Key KI not given

### ISSUE THREAD

**Issue by Endymionls** [DerpyProjectGroup/topaz#461]
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened
- [x] I have checked the commit log to see if the issue has been resolved since my server was last updated
- [X] I have read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [X] I have specified what branch this happens on branch: base

**_Additional Information_** (Steps to reproduce/Expected behavior) **:** 
version 202102_1

Bastok Mission 2-3 San d'Oria > Windurst Path. In Windurst, Kupipi does not give the Dark Key Key Item which then doesn't allow access to the BCNM Dragon fight.

**Comment by CatsEyeXI** [comment]
I do believe I've already opened an issue on this...

However, in the meantime, this is how i patched the issue on my server... in Kupipi.lua...

```
        -- Bastok Mission 2-3 Part II - San d'Oria > Windurst
        elseif currentMission == xi.mission.id.bastok.THE_EMISSARY_WINDURST2 then
            if missionStatus == 7 or missionStatus == 0 then
-- proposed fix
                player:startEvent(242, 1, 1, 1, 1, pNation)
                player:addKeyItem(xi.ki.DARK_KEY)
                player:messageSpecial(ID.text.KEYITEM_OBTAINED, xi.ki.DARK_KEY)
                player:setMissionStatus(player:getNation(), 8)
-- end proposed fix
            elseif missionStatus == 8 then
                player:startEvent(243)
            elseif missionStatus == 9 then
                player:startEvent(244)
            elseif missionStatus == 10 then
                player:startEvent(245)
            end
        else
            player:startEvent(251)
```


and comment out these lines in the next function....

```
entity.onEventFinish = function(player, csid, option)
    if csid == 238 then
        if player:getNation() == xi.nation.BASTOK then
            player:setMissionStatus(player:getNation(), 4)
            player:addKeyItem(xi.ki.SWORD_OFFERING)
            player:messageSpecial(ID.text.KEYITEM_OBTAINED, xi.ki.SWORD_OFFERING)
        end
    elseif csid == 244 then
        player:setMissionStatus(player:getNation(), 10)
--    elseif csid == 242 and player:getNation() == xi.nation.WINDURST then
--    elseif csid == 242 and player:getNation() == xi.nation.BASTOK then
--        player:addKeyItem(xi.ki.DARK_KEY)
--        player:messageSpecial(ID.text.KEYITEM_OBTAINED, xi.ki.DARK_KEY)
--        player:setMissionStatus(player:getNation(), 8)
    elseif csid == 95 then
        player:setMissionStatus(player:getNation(), 1)
        player:addKeyItem(xi.ki.LETTER_TO_THE_CONSULS_WINDURST)
        player:messageSpecial(ID.text.KEYITEM_OBTAINED, xi.ki.LETTER_TO_THE_CONSULS_WINDURST)
```

**Comment by claywar** [comment]
Allowing `missionStatus == 0` to trigger the cutscene for the second half of the mission is absolutely _not_ the way to fix this.  That would allow your players to bypass the first half.

**Comment by claywar** [comment]
Fixed by #462
